assets/src/components/groupedAutocomplete.tsx

```diff
@@ -0,0 +1,731 @@
+          // set the cursor to the first element if:
+          // - the user focused an element in the autocomplete control that is not an option
+          //   (option focus event prevents propagation).
+          // - the cursor is unset or the autocomplete control does not contain the previously focused element.
+          if (
+            cursorLocation === undefined ||
+            autocompleteControlAlreadyHadFocus === false
+          ) {
+            updateCursorLocation(CursorLocationAction.MoveToStart)
+          }
+        }}
+        onBlur={(event) => {
+          // Delete the cursor state if the user's focus exits the autocomplete
+          // control.
+          if (event.currentTarget.contains(event.relatedTarget)) {
+            return
+          }
+
+          updateCursorLocation(CursorLocationAction.DeleteCursor)
+        }}
+        onKeyDown={(e) => {
+          // Handle cursor movement by keyboard.
+          // If the key is not a movement key, allow event to bubble up.
+          switch (e.key) {
+            case "ArrowDown":
+              updateCursorLocation(CursorLocationAction.MoveNext)
+              break
+            case "ArrowUp":
+              updateCursorLocation(CursorLocationAction.MovePrev)
+              break
+            case "Home":
+              updateCursorLocation(CursorLocationAction.MoveToStart)
+              break
+            case "End":
+              updateCursorLocation(CursorLocationAction.MoveToEnd)
+              break
+            default:
+              return
+          }
+          e.preventDefault()
+          e.stopPropagation()
+        }}
+        aria-labelledby={listHeadingId}
+        tabIndex={-1}
+      >
+        {optionGroups.map(({ group: { title, options } }, groupIndex) => (
+          <GroupOptionList key={groupIndex} heading={title}>
+            {options.map(
+              ({ option: { label, onSelectOption } }, optionIndex) => {
+                const selected =
+                  cursorLocation &&
+                  cursorLocation.group === groupIndex &&
+                  cursorLocation.option === optionIndex
+                return (
+                  <li
+                    key={optionIndex}
+                    className="c-autocomplete__option"
+                    role="option"
+                    aria-selected={selected}
+                    onFocus={(e) => {
+                      // Set cursor to this location if focused.
+                      !selected &&
+                        setCursorLocation({
+                          group: groupIndex,
+                          option: optionIndex,
+                        })
+                      e.stopPropagation()
+                    }}
+                    onClick={onSelectOption}
+                    onKeyDown={(e) => {
+                      // Fire `onSelectOption` if enter is pressed
+                      if (!["Enter"].includes(e.key)) {
```

Comment:
there's no reason it can't be, I was just leaving it open in case we ended up needing to add more keys. I think at the time I thought it looked cleaner, but we can update it.

Reply:
Makes sense, I guess it looked slightly weird to me but I have no strong preference in favor of changing it either (although I see you just did so in e5bed05).